File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = set()
    
    def dfs(node):
        stack = [node]
        
        while stack:
            current = stack.pop()
            
            for i in range(n):
                if computers[current][i] == 1 and i not in visited:
                    visited.add(i)
                    stack.append(i)
                    
    for node in range(n):
        if node not in visited:
            dfs(node)
            visited.add(node)
            answer += 1
        
        
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    Set<Integer> visited; 
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        visited = new HashSet<>();
        
        for(int i=0; i<n; ++i){
            if(!visited.contains(i)){
                dfs(i, n, computers);
                ++answer;
            }
        }
        
        return answer;
    }
    
    public void dfs(int start, int n, int[][] computers){
        Deque<Integer> queue = new ArrayDeque<>();
        queue.offerLast(start);       

        while(!queue.isEmpty()){
            int cur = queue.pollFirst();
            visited.add(cur);

            for(int i=0; i<n; i++){
                if(cur!=i && computers[cur][i] == 1 && !visited.contains(i)){
                    queue.add(i);
                }
            }
        }
    }
}